src/test/resources/com/github/checkstyle/patchfilter/PatchFileFromDiffTools/GitDiffPatch.txt

```diff
@@ -0,0 +1,26 @@
+From 901e713030f194cce682dc98b2ae0d3447af9c97 Mon Sep 17 00:00:00 2001
+From: wltan <53135010+wltan@users.noreply.github.com>
+Date: Sun, 29 Mar 2020 13:39:32 +0800
+Subject: [PATCH] spelling: Fix typo in JavadocMethodCheck
+
+---
+ .../tools/checkstyle/checks/javadoc/JavadocMethodCheck.java | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
+index 6faf17d256..e194c6b944 100644
+--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
++++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
+@@ -717,9 +717,9 @@ private static int calculateTagColumn(Matcher javadocTagMatcher,
+     private static List<ExceptionInfo> combineExceptionInfo(List<ExceptionInfo> list1,
+                                                      List<ExceptionInfo> list2) {
+         final List<ExceptionInfo> result = new ArrayList<>(list1);
+-        for (ExceptionInfo expectionInfo : list2) {
+-            if (result.stream().noneMatch(item -> isExceptionInfoSame(item, expectionInfo))) {
+-                result.add(expectionInfo);
++        for (ExceptionInfo exceptionInfo : list2) {
++            if (result.stream().noneMatch(item -> isExceptionInfoSame(item, exceptionInfo))) {
++                result.add(exceptionInfo);
+             }
+         }
+         return result;
```

Comment:
As I understand it is generated by `git format-patch`. It should end with `--` , git version and newlines. Example:
```
From 7c32611c211917da7723512fb05e9a7d84982322 Mon Sep 17 00:00:00 2001
From: Baratali Izmailov <barataliba@gmail.com>
Date: Sun, 14 Jun 2020 13:50:54 +0100
Subject: [PATCH] TestCommit

---
 src/main/java/com/puppycrawl/tools/checkstyle/Checker.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java b/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
index ac872448b..d487d2ffb 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java
@@ -134,7 +134,7 @@ public class Checker extends AutomaticBean implements MessageDispatcher, RootMod
      * Creates a new {@code Checker} instance.
      * The instance needs to be contextualized and configured.
      */
-    public Checker() {
+    public Checker throws RuntimeException () {
         addListener(counter);
         log = LogFactory.getLog(Checker.class);
     }
-- 
2.25.1.windows.1


```

Reply:
on linux:
```
✔ ~/java/github/romani/checkstyle [master|✔] 
$ git format-patch HEAD~1
0001-dependency-bump-archunit-junit5-from-0.13.1-to-0.14..patch

✔ ~/java/github/romani/checkstyle [master|…1] 
$ cat 0001-dependency-bump-archunit-junit5-from-0.13.1-to-0.14..patch 
From f07bcc7fbfa8fc6c4cd8885d2f4f13226425acfa Mon Sep 17 00:00:00 2001
From: "dependabot-preview[bot]"
 <27856297+dependabot-preview[bot]@users.noreply.github.com>
Date: Wed, 3 Jun 2020 13:56:22 +0000
Subject: [PATCH] dependency: bump archunit-junit5 from 0.13.1 to 0.14.1

Bumps [archunit-junit5](https://github.com/TNG/ArchUnit) from 0.13.1 to 0.14.1.
- [Release notes](https://github.com/TNG/ArchUnit/releases)
- [Commits](https://github.com/TNG/ArchUnit/compare/v0.13.1...v0.14.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>
---
 pom.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pom.xml b/pom.xml
index 221c50e..7feed9f 100644
--- a/pom.xml
+++ b/pom.xml
@@ -226,7 +226,7 @@
     <dependency>
       <groupId>com.tngtech.archunit</groupId>
       <artifactId>archunit-junit5</artifactId>
-      <version>0.13.1</version>
+      <version>0.14.1</version>
       <scope>test</scope>
     </dependency>
     <dependency>
-- 
2.7.4

```

Reply:
the same on multi-file path:
```
~/java/github/romani/checkstyle [master|✔] 
$ git format-patch --stdout b53e612a2ab5ff15d14860e252f84c0f343fe93a~1..b53e612a2ab5ff15d14860e252f84c0f343fe93a
From b53e612a2ab5ff15d14860e252f84c0f343fe93a Mon Sep 17 00:00:00 2001
From: nmancus1 <nmancus1
Date: Thu, 4 Jun 2020 11:46:34 -0400
Subject: [PATCH] minor: Add input file for Java14 instanceof with pattern
 matching (#7290)

---
 config/ant-phase-verify.xml                        |  2 +
 .../InputJava14InstanceofWithPatternMatching.java  | 76 ++++++++++++++++++++++
 2 files changed, 78 insertions(+)
 create mode 100644 src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java

diff --git a/config/ant-phase-verify.xml b/config/ant-phase-verify.xml
index d27ffef..e0285ce 100644
--- a/config/ant-phase-verify.xml
+++ b/config/ant-phase-verify.xml
@@ -148,6 +148,8 @@
          <exclude name="**/InputMainFrameModelIncorrectClass.java"/>
          <exclude name="**/InputBeforeExecutionExclusionFileFilterIncorrectClass.java"/>
          <exclude name="**/InputJavaParser.java"/>
+         <!-- until https://github.com/checkstyle/checkstyle/issues/7290 -->
+         <exclude name="**/InputJava14InstanceofWithPatternMatching.java"/>
          <!-- Cannot parse until Java 14 support -->
          <exclude name="**/InputJava14Records.java"/>
         </fileset>
diff --git a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java
new file mode 100644
index 0000000..8fa3eba
--- /dev/null
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14InstanceofWithPatternMatching.java
@@ -0,0 +1,76 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.grammar.java14;
+
+import java.util.Arrays;
+import java.util.Locale;
...... I removed it
+        }
+    }
+}
-- 
2.7.4

```

Reply:
I get this example from https://github.com/checkstyle/checkstyle/commit/901e713030f194cce682dc98b2ae0d3447af9c97.patch

Reply:
Please move all that cases to inputs to make sure we support loading of them , this is required to be done in scope of https://github.com/checkstyle/patch-filters/issues/15

Reply:
done